### PR TITLE
Feat html

### PR DESCRIPTION
**Pull request by parasharrajat**
@Julesssss  Please review.

# Overview

Added HTML copy and pasting support.

### Platform
- [x] Android
- [x] IOS
- [x] Web 

# Test Plan

Added all the tests to the Example repo.

### CODE REVIEW

**Comment by parasharrajat** [comment]
I am not sure where to put the Web implementation. 
1. I see that `react-native-clipboard/clipboard` has Web implementation but it's not good in comparison to `react-native-web`. RN-web  has better implementation.
2. In E/app we are using RN-web for the clipboard. (we didn't have web implementation in this lib at that time).

Let me know what should I do. Here is two options
1. Either replace the SetString on this repo as well as add HTML feature to this fork. replace RN-web with this fork in E/app.

2. Add the setHTML part in E/app directly and keep usages of RN-web.

**Comment by parasharrajat** [comment]
@Julesssss  What do you think?

**Comment by Julesssss** [comment]
Damn, that unsigned commit has reappeared. Would you mind doing [this](https://gist.github.com/mattrobertsky/2072b0a1ab81e824a6a5d38240e20b46) to resign it? (I didn't follow every step, just created a new branch from the unsigned commit, resigned it, then cherrypicked the rest)

UPDATE: Don't worry about this for now, only if we continue with this PR
________

> Let me know what should I do. Here is two options
> 1. Either replace the SetString on this repo as well as add HTML feature to this fork. replace RN-web with this fork in E/app.
> 2. Add the setHTML part in E/app directly and keep usages of RN-web.

To make sure I understand correctly, option one would keep the logic within the Clipboard fork, but we would have to make this repo depend on RN-Web. Option two seems much simpler in comparison and we can keep the RN-Web dependency in the main project?

I'd say two makes the most sense, do you see any issues with that approach? (other than not having the clipboard logic self contained in the forked library)

**Comment by parasharrajat** [comment]
@Julesssss I think you misunderstood option 1. I would not add RN-web dependency. to the fork.

1. RN-web's implementation of `setString` is better than the one this lib provides. My suggestion was to improve the SetString code in this repo that mostly adopts the code from the RN-web and I will add set HTML as well that will reuse most code of the setString.
I think it's fine. Only doubt I had was is it fine to change the code `setString` . This will allow us to move RN-web's clipboard module in E/app to this fork completely.

**Forget that** I didn't think that we won't need the setString at all it would be replaced with `setHTML` in E/app. Pushing changes now.

**Comment by parasharrajat** [comment]
@Julesssss  Ready...

**Comment by Julesssss** [comment]
Ah, yeah I misunderstood. Thanks for explaining. 

The only remaining issue is that an unsigned commit is included with your changes somehow. Would you mind [resigning it](https://github.com/Expensify/clipboard/pull/4#issuecomment-982557877)?

**Comment by parasharrajat** [comment]
Ok. I see that it's technically impossible to keep the fork in sync and sign someone's else commits.
Even if I sign those commits now it will override those commits but then any pull request that is made from the main clipboard lib will have those commits as technically the commits your signed are different from the original commits.

The best way to do it is by directly merging the changes into the master from the main repo.

What do you say?

**Comment by parasharrajat** [comment]
Ok, we are good now @Julesssss .

File: src/Clipboard.ts [hunk 1]
```diff
@@ -83,6 +83,30 @@ export const Clipboard = {
...
+   */
+  setHTML(content: string, text: string) {
+    NativeClipboard.setHTML(content, text);
+  },
+  /**
+   * Get content of HTML type, this method returns a `Promise`, so you can use following code to get clipboard content
+   * ```javascript
+   * async _getContent() {
+   *   var content = await Clipboard.getHTML();
+   * }
```
Comment (Julesssss) [inline src/Clipboard.ts:104]: I guess it doesn't matter too much, but maybe this example could not use async & await, as we don't encourage it in App.

**Review by Julesssss — APPROVED** [review]
I'm not too familiar with TypeScript, but the native code looks good to me.

File: src/Clipboard.ts (hunk 1 above)
Comment (Julesssss) [reply src/Clipboard.ts:104]: Nevermind, all the other functions use the same example.

**Pull request merged**